src/test_stack.py

```diff
@@ -1,66 +1,104 @@
 """Tests for stack class implementation."""
-
 from stack import Stack
 import pytest
 
-EMPTY_LIST = Stack()
+EMPTY_STACK_FOR_PUSH = Stack()
+EMPTY_STACK_FOR_POP = Stack()
+
 POP_LIST = Stack()
 POP_LIST.push('one')
-POP_LIST.push('two')
+POP_LIST.push(2)
 POP_LIST.push('three')
+
 LIST_OF_THREE = Stack()
-LIST_OF_THREE.push('one')
-LIST_OF_THREE.push('two')
-LIST_OF_THREE.push('three')
-REMOVE_LIST = Stack()
-REMOVE_LIST.push('orange')
-REMOVE_LIST.push('apple')
-REMOVE_LIST.push('grape')
-LIST_LIST = Stack([1, 2, 3])
+LIST_OF_THREE.push(3)
+LIST_OF_THREE.push(['one', 'two', 'three'])
+LIST_OF_THREE.push((1, 2, 3))
+
+LIST_LIST = Stack([3, 2, 1])
+
 TUPLE_LIST = Stack((2, 4, 9))
+
 STRING_LIST = Stack('cake')
 
+LEN_OF_THREE = Stack('cake')
+LEN_OF_THREE.push('pie')
+LEN_OF_THREE.push(42)
 
-def test_stack_pop_empty():
-    """Test IndexError when pop method called on empty stack."""
-    with pytest.raises(IndexError):
-        EMPTY_LIST.pop()
+
+def test_stack_init_list():
+    """Test length and head of Stack when instantiated with a list."""
+    assert len(LIST_LIST) == 3
+    assert LIST_LIST._linked_list.head.val == 1
```

Comment:
You're not testing the head of the stack. The stack should have no head. You're testing the head of the internal linked list. In theory this may not map to the actual top of the stack. Test the stack itself. You've already tested the linked list in your linked list tests